include/conversation.php

```diff
@@ -144,222 +144,106 @@ function localize_item(&$item)
-				} else {
-					$post_type = DI::l10n()->t('status');
-				}
-		}
-
-		$plink = '[url=' . $obj['plink'] . ']' . $post_type . '[/url]';
-
-		$bodyverb = '';
-		if ($activity->match($item['verb'], Activity::LIKE)) {
-			$bodyverb = DI::l10n()->t('%1$s likes %2$s\'s %3$s');
-		} elseif ($activity->match($item['verb'], Activity::DISLIKE)) {
-			$bodyverb = DI::l10n()->t('%1$s doesn\'t like %2$s\'s %3$s');
-		} elseif ($activity->match($item['verb'], Activity::ATTEND)) {
-			$bodyverb = DI::l10n()->t('%1$s attends %2$s\'s %3$s');
-		} elseif ($activity->match($item['verb'], Activity::ATTENDNO)) {
-			$bodyverb = DI::l10n()->t('%1$s doesn\'t attend %2$s\'s %3$s');
-		} elseif ($activity->match($item['verb'], Activity::ATTENDMAYBE)) {
-			$bodyverb = DI::l10n()->t('%1$s attends maybe %2$s\'s %3$s');
-		}
-
-		$item['body'] = sprintf($bodyverb, $author, $objauthor, $plink);
-	}
-
-	if ($activity->match($item['verb'], Activity::FRIEND)) {
-
-		if ($item['object-type']=="" || $item['object-type']!== Activity\ObjectType::PERSON) return;
+	/// @todo The following functionality needs to be cleaned up. 
+	if (!empty($item['verb'])) {
+		$activity = DI::activity();
 
-		$Aname = $item['author-name'];
-		$Alink = $item['author-link'];
-
-		$xmlhead="<"."?xml version='1.0' encoding='UTF-8' ?".">";
-
-		$obj = XML::parseString($xmlhead.$item['object']);
-		$links = XML::parseString($xmlhead."<links>".XML::unescape($obj->link)."</links>");
-
-		$Bname = $obj->title;
-		$Blink = "";
-		$Bphoto = "";
-		foreach ($links->link as $l) {
-			$atts = $l->attributes();
-			switch ($atts['rel']) {
-				case "alternate": $Blink = $atts['href']; break;
-				case "photo": $Bphoto = $atts['href']; break;
+		if (stristr($item['verb'], Activity::POKE)) {
+			$verb = urldecode(substr($item['verb'],strpos($item['verb'],'#')+1));
+			if (!$verb) {
+				return;
+			}
+			if ($item['object-type']=="" || $item['object-type']!== Activity\ObjectType::PERSON) {
+				return;
 			}
-		}
-
-		$A = '[url=' . Contact::magicLink($Alink) . ']' . $Aname . '[/url]';
-		$B = '[url=' . Contact::magicLink($Blink) . ']' . $Bname . '[/url]';
-		if ($Bphoto != "") {
-			$Bphoto = '[url=' . Contact::magicLink($Blink) . '][img]' . $Bphoto . '[/img][/url]';
-		}
-
-		$item['body'] = DI::l10n()->t('%1$s is now friends with %2$s', $A, $B)."\n\n\n".$Bphoto;
-
-	}
-	if (stristr($item['verb'], Activity::POKE)) {
-		$verb = urldecode(substr($item['verb'],strpos($item['verb'],'#')+1));
-		if (!$verb) {
-			return;
-		}
-		if ($item['object-type']=="" || $item['object-type']!== Activity\ObjectType::PERSON) {
-			return;
-		}
 
-		$Aname = $item['author-name'];
-		$Alink = $item['author-link'];
+			$Aname = $item['author-name'];
+			$Alink = $item['author-link'];
 
-		$xmlhead = "<" . "?xml version='1.0' encoding='UTF-8' ?" . ">";
+			$xmlhead = "<" . "?xml version='1.0' encoding='UTF-8' ?" . ">";
 
-		$obj = XML::parseString($xmlhead.$item['object']);
+			$obj = XML::parseString($xmlhead.$item['object']);
```

Comment:
Extra tabs added? Is that correct?

Reply:
Yes, because of the added conditional `if (!empty($item['verb'])) {` on line 148.